### PR TITLE
XSH-752 Limit the result size before downloading whole result

### PR DESCRIPTION
**Pull request by Dominik-Sauer**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-sdk/gooddata_sdk/compute/model/execution.py [hunk 1]
```diff
@@ -128,6 +128,16 @@ def as_api_model(self) -> models.AfmExecution:
         return models.AfmExecution(execution=execution, result_spec=result_spec)
 
 
+ResultSize = Tuple[Optional[int], ...]
+
+
+class ResultSizeLimitsExceeded(Exception):
```
Comment (lupko) [inline gooddata-sdk/gooddata_sdk/compute/model/execution.py:134]: please add re-export of this to the top-level __init__.py

File: gooddata-pandas/gooddata_pandas/dataframe.py [hunk 2]
```diff
@@ -229,7 +230,10 @@ def for_insight(self, insight_id: str, auto_index: bool = True) -> pandas.DataFr
         return self.for_items(columns, filter_by=filter_by, auto_index=auto_index)
 
     def for_exec_def(
-        self, exec_def: ExecutionDefinition, label_overrides: LabelOverrides = {}
+        self,
+        exec_def: ExecutionDefinition,
+        label_overrides: LabelOverrides = {},
+        result_size_limits: ResultSize = (),
```
Comment (hkad98) [inline gooddata-pandas/gooddata_pandas/dataframe.py:236]: def for_exec_def(
        self,
        exec_def: ExecutionDefinition,
        label_overrides: LabelOverrides = None,
        result_size_limits: ResultSize = None,
    ) -> Tuple[pandas.DataFrame, BareExecutionResponse]:
if label_overrides is None:
    label_overrides = {}
if result_size_limits is None:
   result_size_limits = ()

**Review by hkad98 — CHANGES_REQUESTED** [review]

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-pandas/gooddata_pandas/dataframe.py (hunk 2 above)
Comment (Dominik-Sauer) [reply gooddata-pandas/gooddata_pandas/dataframe.py:236]: I'm not changing label_overrides, as it's not in the scope of my change. Also, why is the original wrong?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: gooddata-pandas/gooddata_pandas/dataframe.py (hunk 2 above)
Comment (Dominik-Sauer) [reply gooddata-pandas/gooddata_pandas/dataframe.py]: fixed

File: gooddata-pandas/gooddata_pandas/dataframe.py [hunk 3]
```diff
@@ -262,11 +266,20 @@ def for_exec_def(
...
+            ),
             execution.bare_exec_response,
         )
 
-    def for_exec_result_id(self, result_id: str, label_overrides: LabelOverrides = {}) -> pandas.DataFrame:
+    def for_exec_result_id(
+        self,
+        result_id: str,
+        label_overrides: LabelOverrides = {},
+        result_size_limits: ResultSize = (),
```
[the remark this answers is not in the document]
Comment (Dominik-Sauer) [reply gooddata-pandas/gooddata_pandas/dataframe.py]: fixed

**Review by lupko — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-python-sdk-gate-PR140

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-sonarqube-scan-zuul-gate/126/ : SUCCESS in 58s
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-fossa-scanning-zuul-el7/126/ : FAILURE in 12s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-python-sdk-merge-compliance-check/128/ : SUCCESS in 19s

**Pull request merged**